src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AddClassProjectCommand.cs

```diff
@@ -0,0 +1,75 @@
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal class AddClassProjectCommand : AbstractSingleNodeProjectCommand
+    {
+        private readonly IPhysicalProjectTree _projectTree;
+        private readonly IUnconfiguredProjectVsServices _projectVsServices;
+        private readonly SVsServiceProvider _serviceProvider;
+
+        [ImportingConstructor]
+        public AddClassProjectCommand(IPhysicalProjectTree projectTree, IUnconfiguredProjectVsServices projectVsServices, SVsServiceProvider serviceProvider)
+        {
+            Requires.NotNull(projectTree, nameof(IPhysicalProjectTree));
+            Requires.NotNull(projectVsServices, nameof(IUnconfiguredProjectVsServices));
+            Requires.NotNull(serviceProvider, nameof(SVsServiceProvider));
+
+            _projectTree = projectTree;
+            _projectVsServices = projectVsServices;
+            _serviceProvider = serviceProvider;
+        }
+
+        protected override Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string commandText, CommandStatus progressiveStatus)
+        {
+            if (_projectTree.NodeCanHaveAdditions(node))
+            {
+                return GetCommandStatusResult.Handled(commandText, CommandStatus.Enabled);
+            }
+            else
+            {
+                return GetCommandStatusResult.Unhandled;
+            }
+        }
+
+        protected override async Task<bool> TryHandleCommandAsync(IProjectTree node, bool focused, long commandExecuteOptions, IntPtr variantArgIn, IntPtr variantArgOut)
+        {
+            if (!_projectTree.NodeCanHaveAdditions(node))
+            {
+                return false;
+            }
+
+            ConfigurationGeneral projectProperties =
+                            await _projectVsServices.ActiveConfiguredProjectProperties.GetConfigurationGeneralPropertiesAsync().ConfigureAwait(false);
+            Guid guid = new Guid((string)await projectProperties.ProjectGuid.GetValueAsync().ConfigureAwait(false));
+
+            __VSADDITEMFLAGS uiFlags = __VSADDITEMFLAGS.VSADDITEM_AddNewItems | __VSADDITEMFLAGS.VSADDITEM_SuggestTemplateName | __VSADDITEMFLAGS.VSADDITEM_AllowHiddenTreeView;
+
+            IVsAddProjectItemDlg addItemDialog = _serviceProvider.GetService<IVsAddProjectItemDlg, SVsAddProjectItemDlg>();
+            Assumes.Present(addItemDialog);
+
+            string strBrowseLocations = _projectTree.TreeProvider.GetAddNewItemDirectory(node);
+
+            string strFilter = string.Empty;
+            int iDontShowAgain;
+
+            await _projectVsServices.ThreadingService.SwitchToUIThread();
+            HResult res = addItemDialog.AddProjectItemDlg(node.GetHierarchyId(), ref guid, _projectVsServices.VsProject, (uint)uiFlags,
+                Resources.CSharpItemsDirName, Resources.ClassTemplateName, ref strBrowseLocations, ref strFilter, out iDontShowAgain);
```

Comment:
This is in the base-layer - Should it be CSharpItemsDirName? is that value different for VB?